app/controllers/comments_controller.rb

```diff
@@ -1,6 +1,11 @@
 class CommentsController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_post_and_user, only: %i[new create destroy]
+  before_action :set_post_and_user, only: %i[index new create destroy]
+
+  def index
+    @comments = @post.comments
+    render json: @comments
+  end
```

Comment:
- [ ] Kindly consider adding API controllers for your API endpoints. you can take help from this [resource](https://medium.com/@ryanwaldorf/how-to-add-an-api-to-an-existing-rails-app-1d2419ba91bd). I could not get json data from postman from api endpoints.